csharp/source/autowrap/csharp/csharp.d

```diff
@@ -0,0 +1,1061 @@
+                        }
+                        exp ~= "        }" ~ newline;
+                    }
+                    csagg.methods ~= exp;
+                }
+            }
+
+            bool isProperty = false;
+            bool propertyGet = false;
+            bool propertySet = false;
+            foreach(mo; __traits(getOverloads, agg, m)) {
+            static if (cast(bool)(functionAttributes!mo & FunctionAttribute.property)) {
+                isProperty = true;
+                alias returnType = ReturnType!mo;
+                alias paramTypes = Parameters!mo;
+                if (paramTypes.length == 0) {
+                    propertyGet = true;
+                } else {
+                    propertySet = true;
+                }
+            }
+            }
+
+            const olc = __traits(getOverloads, agg, m).length;
+            static if(olc > 0) {
+            if (isProperty) {
+                string prop = string.init;
+                alias propertyType = ReturnType!(__traits(getOverloads, agg, m)[0]);
+                if (isArray!(propertyType)) {
+                    prop = "        public Range<" ~ getCSharpInterfaceType(fullyQualifiedName!propertyType) ~ "> " ~ methodName ~ " { ";
+                } else {
+                    prop = "        public " ~ getCSharpInterfaceType(fullyQualifiedName!propertyType) ~ " " ~ methodName ~ " { ";
+                }
+                if (propertyGet) {
+                    if (isArray!(propertyType)) {
+                        prop ~= "get => new Range<%3$s>(dlang_%1$s(%2$s)); ".format(methodInterfaceName, is(agg == class) ? "DLangPointer" : "ref this", getCSharpInterfaceType(fullyQualifiedName!propertyType));
+                    } else if (is(propertyType == class)) {
+                        prop ~= "get => new %3$s(dlang_%1$s(%2$s)); ".format(methodInterfaceName, is(agg == class) ? "DLangPointer" : "ref this", getCSharpInterfaceType(fullyQualifiedName!propertyType));
+                    } else {
+                        prop ~= "get => dlang_%1$s(%2$s); ".format(methodInterfaceName, is(agg == class) ? "DLangPointer" : "ref this");
+                    }
+                }
+                if (propertySet) {
+                    if (isArray!(propertyType)) {
+                        prop ~= "set => dlang_%1$s(%2$s, value.ToSlice()); ".format(methodInterfaceName, is(agg == class) ? "DLangPointer" : "ref this");
+                    } else if (is(propertyType == class)) {
+                        prop ~= "set => dlang_%1$s(%2$s, value.DLangPointer); ".format(methodInterfaceName, is(agg == class) ? "DLangPointer" : "ref this");
+                    } else {
+                        prop ~= "set => dlang_%1$s(%2$s, value); ".format(methodInterfaceName, is(agg == class) ? "DLangPointer" : "ref this");
+                    }
+                }
+                prop ~= "}" ~ newline;
+                csagg.properties ~= prop;
+            }
+            }
+            }
```

Comment:
The need for avoiding indents is a good sign that some refactoring is in order.

Reply:
I see this as a stylistic question with the handling static if's because static if changes the layout of the code but does not introduce a new runtime scope, so is it really indented? You'll note I only do that with static if's. I can go either way, but to me, it's weirdness serves as  a visual reminder that I am not creating a new scope.

Reply:
Sure, style is subjective.

However, the number of braces is a hint that refactoring is in order. I'm pretty sure it could be easily broken in to functions with nice clear names.